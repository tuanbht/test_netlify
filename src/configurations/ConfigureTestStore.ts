import { configureStore, EnhancedStore } from '@reduxjs/toolkit';
import MockAdapter from 'axios-mock-adapter';
import axiosMiddleware from 'redux-axios-middleware';
import RootReducer from '../reducers';
import axiosClient from './ApiClient';

export const testStore = (initialState = {}): EnhancedStore =>
  configureStore({
    preloadedState: initialState,
    reducer: RootReducer,
    middleware: [axiosMiddleware(axiosClient)],
  });

export const mockAxios = new MockAdapter(axiosClient);
