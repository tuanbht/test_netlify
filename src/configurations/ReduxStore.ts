import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { createBrowserHistory } from 'history';
import axiosMiddleware from 'redux-axios-middleware';
import axiosClient from './ApiClient';
import rootReducer from '../reducers';
import logger from 'redux-logger';

const history = createBrowserHistory();

const middleware = [
  axiosMiddleware(axiosClient),
  ...getDefaultMiddleware({
    serializableCheck: false,
  }),
  logger,
];

const store = configureStore({
  reducer: rootReducer,
  middleware: middleware,
});

export { history, store };
