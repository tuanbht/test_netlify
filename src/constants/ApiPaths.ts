export const API_URL = process.env.REACT_APP_BASE_API;

export const ORDER_DETAILS_PATH = '/orders/:id/payment/';
export const MARK_AS_PAID_PATH = '/orders/:id/mask_as_paid/';
export const CANCEL_ORDER_PATH = '/orders/:id/payment/cancel/';
