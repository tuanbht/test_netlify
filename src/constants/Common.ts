export const VIEW_CRYPTO_TRANSACTION = process.env.REACT_APP_VIEW_CRYPTO_TRANSACTION;
