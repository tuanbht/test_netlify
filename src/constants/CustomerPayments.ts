import React from 'react';
import { ReactComponent as IconBTC } from '../assets/images/icons/icon-btc.svg';
import { ReactComponent as IconETH } from '../assets/images/icons/icon-eth.svg';
import { ReactComponent as IconUSDT } from '../assets/images/icons/icon-usdt.svg';

export const STEPS = {
  CRYPTO: 0,
  WALLET: 1,
  PROCESSING: 2,
  COMPLETE: 3,
};

export interface CryptoCurrencies {
  prefix: string;
  shortName: string;
  fullName: string;
  logo: React.FC<React.SVGProps<SVGSVGElement>>;
  walletAddress: string;
  amount: number;

  setWalletAddress(walletAddress: string): void;
  setAmount(amount: number): void;
}

export class CryptoCurrencies implements CryptoCurrencies {
  static initWithData(
    prefix: string,
    shortName: string,
    fullName: string,
    logo: React.FC<React.SVGProps<SVGSVGElement>>,
  ): CryptoCurrencies {
    const crypto = new CryptoCurrencies();
    crypto.prefix = prefix;
    crypto.shortName = shortName;
    crypto.fullName = fullName;
    crypto.logo = logo;
    return crypto;
  }
  setWalletAddress(walletAddress: string): void {
    this.walletAddress = walletAddress;
  }
  setAmount(amount: number): void {
    this.amount = amount;
  }
}

export const CRYPTO_CURRENCIES = {
  BITCOIN: CryptoCurrencies.initWithData('bitcoin', 'BTC', 'Bitcoin', IconBTC),
  ETHEREUM: CryptoCurrencies.initWithData('ethereum', 'ETH', 'Ethereum', IconETH),
  USDT: CryptoCurrencies.initWithData('usdt', 'USDT', 'USD Tether', IconUSDT),
};
