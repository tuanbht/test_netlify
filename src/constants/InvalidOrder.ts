export const ORDER_CANCELLED_CONTENT = {
  title: 'Order cancelled',
  subtitle: 'This order has been cancelled and is no longer available. To place a new order, contact your Merchant.',
};

export const ORDER_EXPIRED_CONTENT = {
  title: 'Order expired',
  subtitle: 'This order expired and is no longer available. To place a new order, contact your Merchant.',
};
