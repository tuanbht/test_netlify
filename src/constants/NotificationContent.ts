export const ORDER_CANCELLED_CONTENT = {
  title: 'Order cancelled',
  subtitle: 'This order has been cancelled and is no longer available. To place a new order, contact your Merchant.',
};

export const ORDER_EXPIRED_CONTENT = {
  title: 'Order expired',
  subtitle: 'This order expired and is no longer available. To place a new order, contact your Merchant.',
};
export const ROOT_PAGE_CONTENT = {
  title: 'Welcome to Paloma!',
  subtitle:
    "To pay for your order with Crypto, follow the link sent via text message or ask your merchant to initiate the payment process if you haven't yet received a text.",
};
