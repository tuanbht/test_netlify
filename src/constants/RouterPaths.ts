export const ORDER_PATH = '/:orderId/:token';
export const ORDER_EXPIRED_PATH = '/order-expired';
export const ORDER_CANCELLED_PATH = '/order-cancelled';
export const NOT_FOUND_PATH = '/';
