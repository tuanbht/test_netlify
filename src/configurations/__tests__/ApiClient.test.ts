import axiosClient from '../ApiClient';
import { store } from '../ReduxStore';
import faker from 'faker';
import { OrderDetails } from '../../constants/CustomerPayments';
import { mockAxios } from '../ConfigureTestStore';
import { NOT_FOUND_PATH } from '../../constants/RouterPaths';

describe('ApiClient', () => {
  const token = faker.random.uuid();

  describe('axiosClient', () => {
    it('returns default with customized headers', () => {
      expect(axiosClient.defaults.headers).toMatchSnapshot();
    });
  });

  describe('axios request header', () => {
    beforeEach(() => {
      mockAxios.onAny().reply(200);
    });

    describe('payment token is stored', () => {
      beforeEach(() => {
        jest.spyOn(store, 'getState').mockReturnValue({
          OrderDetails: new OrderDetails(),
          Credential: {
            orderId: 0,
            token,
          },
        });
      });

      it('makes request with payment token header', async () => {
        const response = await axiosClient.get('/');
        expect(response.config.headers['Payment-Token']).toEqual(token);
      });
    });

    describe('payment token is not stored', () => {
      beforeEach(() => {
        jest.spyOn(window.location, 'assign').mockReturnValue();

        jest.spyOn(store, 'getState').mockReturnValue({
          OrderDetails: new OrderDetails(),
          Credential: {
            orderId: 0,
            token: '',
          },
        });
      });

      it('redirects to not found page', async () => {
        const response = await axiosClient.get('/');
        expect(response.config.headers['Payment-Token']).toBeUndefined();
        expect(window.location.assign).toBeCalledWith(NOT_FOUND_PATH);
      });
    });
  });
});
