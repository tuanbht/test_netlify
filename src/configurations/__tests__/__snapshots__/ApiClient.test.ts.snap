// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiClient axiosClient returns default with customized headers 1`] = `
Object {
  "Accept": "application/vnd.api+json",
  "Content-Type": "application/vnd.api+json",
  "common": Object {
    "Accept": "application/json, text/plain, */*",
  },
  "delete": Object {},
  "get": Object {},
  "head": Object {},
  "patch": Object {
    "Content-Type": "application/x-www-form-urlencoded",
  },
  "post": Object {
    "Content-Type": "application/x-www-form-urlencoded",
  },
  "put": Object {
    "Content-Type": "application/x-www-form-urlencoded",
  },
}
`;
